src/commands/removeGithubThread.ts

```diff
@@ -0,0 +1,52 @@
+import { Command } from "@yuudachi/framework";
+import type { InteractionParam, ArgsParam, CommandMethod } from "@yuudachi/framework/types";
+import { PermissionFlagsBits } from "discord.js";
+import { NormalGitHubUrlRegex, GitHubUrlLinesRegex } from "../functions/github/regex.js";
+import type { RemoveGithubThread } from "../interactions/context/removeGithubThread.js";
+
+export default class extends Command<typeof RemoveGithubThread> {
+	public constructor() {
+		super(["Remove Github Thread"]);
+	}
+	private matchChannelName(name: string): boolean {
+		return name === "GitHub Lines for this message";
+	}
+	public override async messageContext(
+		interaction: InteractionParam<CommandMethod.MessageContext>,
+		args: ArgsParam<typeof RemoveGithubThread>,
+	): Promise<void> {
+		if (interaction.channel?.isThread()) {
+			const starterMessage = await interaction.channel.fetchStarterMessage();
+			if (!this.matchChannelName(interaction.channel?.name) && interaction.channel.ownerId !== interaction.client.user.id) {
+				interaction.reply({ content: "This is not github thread.", ephemeral: true });
+				return;
+			}
+			if ( 
+				interaction.user.id !== starterMessage?.author.id &&
+				!interaction.member.permissions.has(PermissionFlagsBits.ManageThreads)
+			) {
+				interaction.reply({ content: "You are not able to do it.", ephemeral: true });
+				return;
+			}
+			interaction.channel.delete("Remove Github Lines command!");
```

Comment:
Maybe reply the interaction with `This thread will be deleted in X seconds` and add a timer

```suggestion
			await interaction.channel.delete("Remove Github Lines command!");
```

Reply:
Cool visual  feature 
but useless when using it

Reply:
The thing is, you will let the interaction fail?

Reply:
But the channel is deleted